lib/deployinator/controller.rb

```diff
@@ -107,10 +107,12 @@ def run(options)
       deploy_instance = deploy_class.new(options)
       deploy_instance.register_plugins(options[:stack])
 
-      locked = deploy_instance.lock_pushes(options[:stack], options[:username], options[:method])
+      if requires_lock
+        locked = deploy_instance.lock_pushes(options[:stack], options[:username], options[:method])
```

Comment:
all the references to `locked` are within the if clause (there aren't any others in the body of the `Controller#run` method), so I think we're ok.